src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -15,22 +19,28 @@
 
     private final boolean isAddTask;
     private Task newTask = null;
+    private String targetModuleName = null;
     private Module newModule = null;
 
-    public AddCommand(String name, String description, boolean isTask, String estimatedWorkingTime) {
-        if (isTask) {
-            newTask = new Task(name, description, estimatedWorkingTime);
-            isAddTask = true;
-        } else {
-            newModule = new Module(name, description);
-            isAddTask = false;
-        }
+    public AddCommand(String name, String description, String estimatedWorkingTime, String taskModule) {
+        newTask = new Task(name, description, estimatedWorkingTime);
+        targetModuleName = taskModule;
+        isAddTask = true;
+    }
+
+    public AddCommand(String moduleName, String moduleDescription) {
+        newModule = new Module(moduleName, moduleDescription);
+        isAddTask = false;
     }
```

Comment:
Comparing with the original constructor, this confuses me a little bit. If we view the data type signature of the two constructor, one is AddCommand(String, String) and the other is AddCommand(String, String, String). However the first one is to add task while the second is to add module. This throw the decision of types to outsider classes, and seems less abstract and encapsulated.

Reply:
Let's have an enum describing the type of object being created by the command and make it one of the parameters. That way we won't run into problems if we add more task types.

Reply:
okk